src/pages/Quiz/Quiz.jsx

```diff
@@ -0,0 +1,182 @@
+  const [counter, setCounter] = useState(0);
+  const [score, setScore] = useState(0);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [options, setOptions] = useState([]);
+  const incrementCounter = () => setCounter((current) => current + 1);
+  const [timer, setTimer] = useState(30);
+  const id = useRef(null);
+  const clear = () => {
+    window.clearInterval(id.current);
+  };
+  useEffect(() => {
+    id.current = window.setInterval(() => {
+      setTimer((time) => time - 1);
+    }, 1000);
+    return () => clear();
+  }, [counter]);
+
+  useEffect(() => {
+    if (timer === 0) {
+      clear();
+      counter + 1 === quantity ? handleSubmit() : handleNext();
+    }
+  }, [timer]);
+
+  useEffect(() => {
+    if (!quizState.difficulty && !quizState.quantity) {
+      makeToast(
+        'Please select difficulty and quantity first to proceed with the quiz',
+        'error'
+      );
+      navigate(`/rules/${quizId}`);
+      return;
+    }
+    (async () => {
+      try {
+        setLoading(true);
+        const { quiz } = await getSelectedQuiz(quizId);
+        setSelectedQuiz(quiz);
+        const data = getRandomQuestions(quiz[difficulty], quantity);
+        setSelectedQuestions(data);
+        const optionsShuffled = shuffleOptions([
+          ...data[counter]?.incorrect_answers,
+          data[counter]?.correct_answer,
+        ]);
+        setOptions([...optionsShuffled]);
+      } catch (error) {
+        console.log(error.message);
+      } finally {
+        setLoading(false);
+      }
+    })();
+  }, []);
+  useEffect(() => {
+    if (selectedQuestions.length > 0) {
+      const optionsShuffled = shuffleOptions([
+        ...selectedQuestions[counter]?.incorrect_answers,
+        selectedQuestions[counter]?.correct_answer,
+      ]);
+      setOptions([...optionsShuffled]);
+    }
+  }, [counter]);
+  const handleOptionClick = (option) => setSelectedOption(option);
+
+  const addToAttempted = () => {
+    const attempted = {
+      ...selectedQuestions[counter],
+      attempted: selectedOption,
+    };
+    setAttemptedQuestions((a) => [...a, attempted]);
+  };
+
+  const handleNext = () => {
+    if (selectedOption === selectedQuestions[counter]?.correct_answer) {
+      setScore((s) => s + 10);
+    }
+    addToAttempted();
+    setSelectedOption('');
+    incrementCounter();
+    setTimer(30);
+  };
+  const handleSubmit = () => {
+    let finalScore = score;
+    if (selectedOption === selectedQuestions[counter]?.correct_answer) {
+      setScore((s) => s + 10);
+      finalScore = finalScore + 10;
+    }
+    const attempted = {
+      ...selectedQuestions[counter],
+      attempted: selectedOption,
+    };
+    makeToast(`You Scored ${finalScore} points!`, 'success');
+    quizDispatch({
+      type: 'LOAD_RESULTS',
+      payload: {
+        attemptedQuestions: [...attemptedQuestions, attempted],
+        finalScore,
+      },
+    });
+    navigate('/');
+  };
+
+  return (
```

Comment:
Since you are using ternary operators in the return statement, for better readability -

Recently I learned a way to make our code inside return look clean. What you can do is, before the return statement where we write all the logic you can make components. For example -

export default func XYZ(){

const yourComponent = (

write all your jsx here

)

return (

yourComponent ? yourComponent2 : yourComponent3

)
}

Reply:
Thanks for the suggestion